lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -458,13 +475,16 @@ def _download_file(url, b_path, expected_hash, validate_certs, token=None, timeo
     display.display("Downloading %s to %s" % (url, to_text(b_tarball_dir)))
     # NOTE: Galaxy redirects downloads to S3 which rejects the request
     # NOTE: if an Authorization header is attached so don't redirect it
-    resp = open_url(
```

Comment:
Out-of-the-scope note:
This leaves an unclosed resource open and should be converted to a CM in a future refactoring.